### PR TITLE
[PLAT-493][Gates] Allow project files to be embedded/rendered in the wiki

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose 

Allows files hosted on the OSF to be rendered via the MFR and embed into the wiki markdown.

## Changes

- Adds an mfr service to the the other renderers.
- Adds tests to the test.js file*.
- Adds doc info to the readme.

* These tests can't be done via markdown-it-testgen because they require a random id which markdown-it-testgen can't handle.

## Side Effects

None that I know of.

## QA Notes

See https://github.com/CenterForOpenScience/osf.io/pull/8133

## Ticket

https://openscience.atlassian.net/browse/PLAT-493

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -29,6 +30,14 @@ function preziParser(url) {
   return match ? match[1] : url;
 }
 
+// TODO: Write regex for staging and local servers.
+const mfrRegex = /^http(?:s?):\/\/(?:www\.)?mfr\.osf\.io\/render\?url=http(?:s?):\/\/osf\.io\/([a-zA-Z0-9]{1,5})\/\?action=download/;
```
Comment (brianjgeiger) [inline index.js:34]: I think this is overly restrictive. You should just check for something in the range of `https://osf.io/<guid>/` (or the subdomain variants `https://*.osf.io/<guid>/` and throw away the rest. That would make it easy to get the subdomain as well.

File: test/test.js [hunk 2]
```diff
@@ -9,3 +14,63 @@ describe('markdown-it-video', function requireMarkdownIt() {
   }).use(require('../'));
   generate(path.join(__dirname, 'fixtures/video.txt'), md);
 });
+
+// Because the mfr iframe requires a random id these tests cannont be part of
```
Comment (brianjgeiger) [inline test/test.js:19]: That's a shame.

**Review by brianjgeiger — APPROVED** [review]
Some comments for future improvements, but not for this ticket.

**Comment by brianjgeiger** [comment]
Test failures. Be sure to run eslint before committing.

```
/home/travis/build/brianjgeiger/markdown-it-video/index.js
  128:14  error  Strings must use singlequote  quotes
  128:74  error  Strings must use singlequote  quotes
  128:93  error  Missing semicolon             semi
  165:41  error  Missing trailing comma        comma-dangle
/home/travis/build/brianjgeiger/markdown-it-video/test/test.js
   5:16  warning  Unexpected unnamed function                func-names
   5:24  error    Missing space before function parentheses  space-before-function-paren
   7:2   error    Missing semicolon                          semi
  29:63  warning  Unexpected unnamed function                func-names
  35:62  warning  Unexpected unnamed function                func-names
  41:77  warning  Unexpected unnamed function                func-names
  47:78  warning  Unexpected unnamed function                func-names
  53:83  warning  Unexpected unnamed function                func-names
  59:62  warning  Unexpected unnamed function                func-names
  65:78  warning  Unexpected unnamed function                func-names
  71:83  warning  Unexpected unnamed function                func-names
```

**Comment by Johnetordoff** [comment]
@brianjgeiger passing tests now should be gtg.

**Pull request merged**